nova_vm/src/ecmascript/builtins/structured_data/data_view_objects/data_view_prototype.rs

```diff
@@ -248,184 +249,364 @@ impl DataViewPrototype {
         Ok(Number::from(SmallInteger::try_from(o.byte_offset(agent) as i64).unwrap()).into_value())
     }
 
+    /// ### [25.3.4.5 DataView.prototype.getBigInt64 ( byteOffset \[ , littleEndian \] )](https://tc39.es/ecma262/#sec-dataview.prototype.getbigint64)
     fn get_big_int64(
-        _agent: &mut Agent,
-        _gc: GcScope<'_, '_>,
-        _this_value: Value,
-        _arguments: ArgumentsList,
+        agent: &mut Agent,
+        gc: GcScope<'_, '_>,
+        this_value: Value,
+        arguments: ArgumentsList,
     ) -> JsResult<Value> {
-        todo!()
+        let byte_offset = arguments.get(0);
+        let little_endian = arguments.get(1);
+        // 1. Let v be the this value.
+        // 2. Return ? GetViewValue(v, byteOffset, littleEndian, bigint64).
+        get_view_value::<i64>(agent, gc, this_value, byte_offset, little_endian)
     }
 
+    /// ### [25.3.4.6 DataView.prototype.getBigUint64 ( byteOffset \[ , littleEndian \] )](https://tc39.es/ecma262/#sec-dataview.prototype.getbiguint64)
     fn get_big_uint64(
-        _agent: &mut Agent,
-        _gc: GcScope<'_, '_>,
-        _this_value: Value,
-        _: ArgumentsList,
+        agent: &mut Agent,
+        gc: GcScope<'_, '_>,
+        this_value: Value,
+        arguments: ArgumentsList,
     ) -> JsResult<Value> {
-        todo!()
+        let byte_offset = arguments.get(0);
+        let little_endian = arguments.get(1);
+        // 1. Let v be the this value.
+        // 2. Return ? GetViewValue(v, byteOffset, littleEndian, biguint64).
+        get_view_value::<u64>(agent, gc, this_value, byte_offset, little_endian)
     }
 
+    /// ### [25.3.4.7 DataView.prototype.getFloat32 ( byteOffset \[ , littleEndian \] )](https://tc39.es/ecma262/#sec-dataview.prototype.getfloat32)
     fn get_float32(
-        _agent: &mut Agent,
-        _gc: GcScope<'_, '_>,
-        _this_value: Value,
-        _: ArgumentsList,
+        agent: &mut Agent,
+        gc: GcScope<'_, '_>,
+        this_value: Value,
+        arguments: ArgumentsList,
     ) -> JsResult<Value> {
-        todo!()
+        let byte_offset = arguments.get(0);
+        // 2. If littleEndian is not present, set littleEndian to false.
+        let little_endian = if arguments.len() > 1 {
```

Comment:
**thought:** Because `to_boolean` cannot throw any errors, we could perform that transform here and call `get_view_value` with an actual bool.